typescript/sdk/src/warp/WarpCore.ts

```diff
@@ -332,16 +333,42 @@ export class WarpCore {
     const providerType = TOKEN_STANDARD_TO_PROVIDER_TYPE[token.standard];
     const hypAdapter = token.getHypAdapter(this.multiProvider, destinationName);
 
-    if (await this.isApproveRequired({ originTokenAmount, owner: sender })) {
-      this.logger.info(`Approval required for transfer of ${token.symbol}`);
+    const [isApproveRequired, isRevokeApprovalRequired] = await Promise.all([
+      this.isApproveRequired({
+        originTokenAmount,
+        owner: sender,
+      }),
+      hypAdapter.isRevokeApprovalRequired(
```

Comment:
why does isApproveRequired use getAdapter and isRevokeApprovalRequired use getHypAdapter?

whats the diff here?

Reply:
Because in transfer remote we are interacting with a HypToken, so I want to be sure that the adapter we are using is for only `HyptToken`s while the `getAdapter` might return an adapter that is not for a HypToken, which would be wrong in this context. getHypAdapter would fail if let's say that the provided token is a simple native or ERC20 one